src/utils/secure.filter.ts

```diff
@@ -30,22 +33,36 @@ export const getFilterPage = (req: Request, res: Response, next: NextFunction, t
   }
 };
 
-export const postFilterPage = (req: Request, res: Response, next: NextFunction, isSecureRegisterYes: string, isSecureRegisterNo: string): void => {
+export const postFilterPage = async (req: Request, res: Response, next: NextFunction, isSecureRegisterYes: string, isSecureRegisterNo: string): Promise<void> => {
   try {
+
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
-    const isSecureRegister = req.body[IsSecureRegisterKey];
 
-    setExtraData(req.session, { ...getApplicationData(req.session), [IsSecureRegisterKey]: isSecureRegister });
+    const appData: ApplicationData = getApplicationData(req.session);
+    const isSecureRegister = (req.body[IsSecureRegisterKey]).toString();
+    appData[IsSecureRegisterKey] = isSecureRegister;
+    const session = req.session as Session;
+
+    let nextPageUrl: string = "";
 
-    let nextPageUrl: any;
-    if (isSecureRegister === '1') {
+    if (isSecureRegister === "1") {
       nextPageUrl = isSecureRegisterYes;
-    } else if (isSecureRegister === '0') {
+    }
+    if (isSecureRegister === "0") {
       nextPageUrl = isSecureRegisterNo;
+      if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)) {
+        if (appData[Transactionkey] && appData[OverseasEntityKey]) {
+          await updateOverseasEntity(req, session, appData);
+        } else {
+          throw new Error("Invalid request");
```

Comment:
The error message doesn't provide much information. Could replace with something like
`Error: Secure Filter cannot update Overseas Entity record - transaction_id or overseas_entity_id is missing`

Reply:
Updated.